libraries/abstractions/pkcs11/mbedtls/iot_pkcs11_mbedtls.c

```diff
@@ -899,13 +894,13 @@ static CK_RV prvEcKeyAttParse( CK_ATTRIBUTE_PTR pxAttribute,
     }
 
     /* private EC key attributes. */
-    if( ( xResult != CKR_OK ) && ( xIsPrivate == CK_TRUE ) )
+    if( ( xResult == CKR_ATTRIBUTE_VALUE_INVALID ) && ( xIsPrivate == CK_TRUE ) )
```

Comment:
Are you intentionally making this logic change?

Reply:
yes, the issue was it would override error messages from an earlier failure, this default case indicates that nothing was caught by the above switch statement

Reply:
ok

Reply:
This approach seems risky from the start, though, because we're attempting to parse the data as a certain format without explicitly checking the attribute type. Safer would be to move these blocks into the switch/case above. That would also make this function easier to review and maintain, since it makes for a more typical structure.

As part of that, xResult could for example be initialized as CKR_OK. Then if you fall into the default switch/case, set it to ATTR_VALUE_INVALID.

Reply:
I had to split this function up to due to the GNU complexity requirement. The switch statement contains too much logic if its all in one. I'm more than happy to combine all the switches into one, as i agree it is easier to read, but i will need to get an exception for the GNU complexity requirement.

Reply:
I have refactored this switch statement to be more clear